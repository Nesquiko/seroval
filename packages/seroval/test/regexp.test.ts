import { describe, it, expect } from 'vitest';
import {
  deserialize,
  fromJSON,
  serialize,
  serializeAsync,
  toJSON,
  toJSONAsync,
} from '../src';

describe('RegExp', () => {
  describe('serialize', () => {
    it('supports RegExp', () => {
      const example = /[a-z0-9]+/i;
      const result = serialize(example);
      expect(result).toMatchSnapshot();
      const back = deserialize<RegExp>(result);
      expect(back).toBeInstanceOf(RegExp);
      expect(String(back)).toBe(String(example));
    });
  });
  describe('serializeAsync', () => {
    it('supports RegExp', async () => {
      const example = /[a-z0-9]+/i;
      const result = await serializeAsync(Promise.resolve(example));
      expect(result).toMatchSnapshot();
      const back = await deserialize<Promise<RegExp>>(result);
      expect(back).toBeInstanceOf(RegExp);
      expect(String(back)).toBe(String(example));
    });
  });
  describe('toJSON', () => {
    it('supports RegExp', () => {
      const example = /[a-z0-9]+/i;
      const result = toJSON(example);
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = fromJSON<RegExp>(result);
      expect(back).toBeInstanceOf(RegExp);
      expect(String(back)).toBe(String(example));
    });
  });
  describe('toJSONAsync', () => {
    it('supports RegExp', async () => {
      const example = /[a-z0-9]+/i;
      const result = await toJSONAsync(Promise.resolve(example));
      expect(JSON.stringify(result)).toMatchSnapshot();
      const back = await fromJSON<Promise<RegExp>>(result);
      expect(back).toBeInstanceOf(RegExp);
      expect(String(back)).toBe(String(example));
    });
  });
});
