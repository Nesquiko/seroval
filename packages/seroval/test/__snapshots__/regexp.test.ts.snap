// Vitest Snapshot v1

exports[`RegExp > serialize > supports RegExp 1`] = `"/[a-z0-9]+/i"`;

exports[`RegExp > serializeAsync > supports RegExp 1`] = `"Promise.resolve(/[a-z0-9]+/i)"`;

exports[`RegExp > toJSON > supports RegExp 1`] = `"{\\"t\\":{\\"t\\":12,\\"i\\":0,\\"c\\":\\"[a-z0-9]+\\",\\"m\\":\\"i\\"},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;

exports[`RegExp > toJSONAsync > supports RegExp 1`] = `"{\\"t\\":{\\"t\\":18,\\"i\\":0,\\"f\\":{\\"t\\":12,\\"i\\":1,\\"c\\":\\"[a-z0-9]+\\",\\"m\\":\\"i\\"}},\\"r\\":0,\\"i\\":false,\\"f\\":8191,\\"m\\":[]}"`;
